[Test Proxy] Guidance for expected DNS lookup failures (#23966)
Azure/azure-sdk-for-python: doc/dev/test_proxy_troubleshooting.md

## KeyError during container startup

Try updating your machine's version of Docker. Older versions of Docker may not return a status to indicate whether or
not the proxy container is running, which the [proxy_startup.py][proxy_startup] script needs to determine.

## ConnectionError during test startup

For example, you may see a `requests.exceptions.ConnectionError` when trying to contact URL `/Info/Available`. This
means that the test proxy tool wasn't started up properly, so requests to the tool are failing. Make sure Docker is
installed and is up to date, and ensure that Linux containers are being used.

## Different error than expected when using proxy

> Some tests intentionally trigger exceptions in order to validate error behavior. There's a known case where the

Reviewer: ```suggestion
Some tests intentionally trigger exceptions in order to validate error behavior. There's are a few known cases where the
```

exception returned will be different when using the test proxy vs. when sending requests to the service directly.

> This happens in the case of a DNS lookup failure, which can occur when trying to contact a nonexistent endpoint.

Reviewer: ```suggestion
Once such occurrence is in the case of a DNS lookup failure, which can occur when trying to contact a nonexistent endpoint.
```
